src/dtls13.c

```diff
@@ -0,0 +1,2432 @@
+        sizeof(WOLFSSL_METHOD), heap, DYNAMIC_TYPE_METHOD);
+
+    if (!method)
+        return NULL;
+
+    InitSSL_Method(method, MakeDTLSv1_3());
+#if !defined(NO_OLD_TLS) && !defined(WOFLSSL_NO_TLS12)
+    method->downgrade = 1;
+#endif
+    method->side = WOLFSSL_SERVER_END;
+
+    return method;
+}
+
+static inline void Dtls13ToBigEndian(word64 in, byte *out)
+{
+    word64 *_out = (word64*)out;
+#ifdef BIG_ENDIAN_ORDER
+    *_out = in;
+#else
+    *_out = ByteReverseWord64(in);
+#endif /* BIG_ENDIAN_ORDER */
+}
+
+static inline void Dtls13FromBigEndian(const byte *in, word64* out)
+{
+    word64 *_in = (word64*)in;
+#ifdef BIG_ENDIAN_ORDER
+    *_out = *_in;
+#else
+    *out = ByteReverseWord64(*_in);
+#endif /* BIG_ENDIAN_ORDER */
+}
+
+int Dtls13RlAddPlaintextHeader(
+    WOLFSSL *ssl, byte *out, enum ContentType content_type, size_t length)
+{
+    Dtls13RecordPlaintextHeader *hdr;
+    word32 seq[2];
+    int ret;
+
+    hdr = (Dtls13RecordPlaintextHeader *)out;
+    hdr->contentType = content_type;
+    hdr->legacyVersionRecord.major = DTLS_MAJOR;
+    hdr->legacyVersionRecord.minor = DTLSv1_2_MINOR;
+
+    ret = Dtls13GetSeq(ssl, CUR_ORDER, seq, 1);
+    if (ret != 0)
+        return ret;
+
+    /* seq[0] combines the epoch and 16 MSB of sequence number. We write on the
+       epoch field and will overflow to the first two bytes of the sequence
+       number */
+    c32toa(seq[0], hdr->epoch);
+    c32toa(seq[1], &hdr->sequenceNumber[2]);
+
+    c16toa(length, hdr->length);
+
+    return 0;
+}
+
+static int Dtls13HandshakeAddHeaderFrag(WOLFSSL *ssl, byte *output,
+    enum HandShakeType msg_type, word32 frag_offset, word32 frag_length,
+    word32 msg_length)
+{
+    Dtls13HandshakeHeader *hdr;
+
+    hdr = (Dtls13HandshakeHeader *)output;
+
+    hdr->msg_type = msg_type;
+    c32to24((word32)msg_length, hdr->length);
+    c16toa(ssl->keys.dtls_handshake_number, hdr->messageSeq);
+
+    c32to24(frag_offset, hdr->fragmentOffset);
+    c32to24(frag_length, hdr->fragmentLength);
+
+    return 0;
+}
+
+static int Dtls13TypeIsEncrypted(enum HandShakeType hs_type)
+{
+    switch (hs_type) {
+    case hello_request:
+    case hello_verify_request:
+    case client_hello:
+    case server_hello:
+        return 0;
+    case encrypted_extensions:
+    case session_ticket:
+    case end_of_early_data   :
+    case hello_retry_request :
+    case certificate         :
+    case server_key_exchange :
+    case certificate_request :
+    case server_hello_done   :
+    case certificate_verify  :
+    case client_key_exchange :
+    case finished            :
+    case certificate_status  :
+    case key_update          :
+    case change_cipher_hs    :
+    case message_hash        :
+    case no_shake            :
+        return 1;
+    }
+
+    /* abide compilers */
+    return 0;
+}
+
+static int Dtls13GetRnMask(
+    WOLFSSL *ssl, const byte *ciphertext, byte *mask, enum rnDirection dir)
```

Comment:
Put argument definitions on line above until overflow.

Reply:
Did just for this for now. Should I apply that to every function? (I was letting the `clangd-formatter` take the decision)

Reply:
Unfortunately formatters don't follow our rules. But sometimes they have options that will work.
I would like to see all functions match our convention of filling up the line and continue on next.

If you find options that make the code look like we want, let us know!
It would be good to be able to run a formatter over all the code to fix the formatting everywhere!